### PR TITLE
[Helix] Add new osx11 helix queue to matrix

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

File: .azure/pipelines/helix-matrix.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # We only want to run full helix matrix on master
-pr: none
+# Run PR validation on all branches
+pr:
+  autoCancel: true
+  branches:
+    include:
+    - '*'
```
Comment (HaoK) [inline .azure/pipelines/helix-matrix.yml:7]: revert

**Comment by HaoK** [comment]
Add OSX11 to helix matrix, also add special "All.OSX" queue name to signal skipping all osx queues since specifying them all was super clunky

File: src/Testing/src/xunit/SkipOnHelixAttribute.cs [hunk 2]
```diff
@@ -42,6 +42,19 @@ public string SkipReason
...
+        
+        private bool ShouldSkip()
+        {
+            if (Queues == null) 
+            {
+                return true;
+            }
+            if (Queues.Contains("All.OSX")) 
+            {
+                return GetTargetHelixQueue().ToLowerInvariant().Contains("osx"));
```
Comment (dougbu) [inline src/Testing/src/xunit/SkipOnHelixAttribute.cs]:
```suggestion
                return GetTargetHelixQueue().ToLowerInvariant().Contains("osx");
```

File: src/Testing/src/xunit/SkipOnHelixAttribute.cs [hunk 3]
```diff
@@ -42,6 +42,19 @@ public string SkipReason
...
+            }
+            return Queues.ToLowerInvariant().Split(';').Contains(GetTargetHelixQueue().ToLowerInvariant();
```
Comment (dougbu) [inline src/Testing/src/xunit/SkipOnHelixAttribute.cs]:
```suggestion
            return Queues.ToLowerInvariant().Split(';').Contains(GetTargetHelixQueue().ToLowerInvariant());
```

nit: Suggest adding a variable set just above containing `GetTargetHelixQueue().ToLowerInvariant()`.

File: src/Testing/src/xunit/SkipOnHelixAttribute.cs (hunk 2 above)
Comment (dougbu) [reply src/Testing/src/xunit/SkipOnHelixAttribute.cs]: Missed the Refresh notification😺

File: src/Testing/src/xunit/SkipOnHelixAttribute.cs [hunk 4]
```diff
@@ -42,6 +42,19 @@ public string SkipReason
...
             }
         }
+        
+        private bool ShouldSkip()
+        {
+            if (Queues == null) 
+            {
+                return true;
+            }
+            if (Queues.Contains("All.OSX"))
```
Comment (dougbu) [inline src/Testing/src/xunit/SkipOnHelixAttribute.cs:52]: nit: Suggest adding a variable set above this line containing `GetTargetHelixQueue().ToLowerInvariant()`. Also slightly prefer a blank line after the previous block.

**Review by dougbu — APPROVED** [review]

**Comment by HaoK** [comment]
@Tratcher are there any known issues with AppCanHandleClientAbortingConnectionMidRequest on OSX?  I'm seeing this fail consistently on the new OSX11 queue we are trying to enable

**Pull request merged**